### ISSUE TITLE
_create_mask helper function not checking for values correctly 

### ISSUE THREAD

**Issue by mirob9363** [earthlab/earthpy#276]
**Describe the bug**
When I am trying to call `mask_pixels`, I am getting the following error:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-20-0748834153dc> in <module>()
      1 # Call the earthpy mask function using pixel QA layer
----> 2 landsat_pre_cl_free = em.mask_pixels(landsat_pre, landsat_qa, all_masked_values)

~\AppData\Local\Continuum\anaconda3\envs\earth-analytics-python\lib\site-packages\earthpy\mask.py in mask_pixels(arr, mask_arr, vals)
    224 
    225     if vals:
--> 226         cover_mask = _create_mask(mask_arr, vals)
    227     else:
    228         # Check to make sure the mask_arr is a boolean

~\AppData\Local\Continuum\anaconda3\envs\earth-analytics-python\lib\site-packages\earthpy\mask.py in _create_mask(mask_arr, vals)
    123         raise ValueError(
    124             """The values provided for the mask do not occur
--> 125             in your mask array."""
    126         )
    127 

ValueError: The values provided for the mask do not occur
            in your mask array.
```

**To Reproduce**
use this code (data from the cold-springs-fire) to create the issue:
```

import earthpy as et
import earthpy.spatial as es
from glob import glob
import rasterio as rio
from rasterio.plot import plotting_extent
import os
import earthpy.mask as em
data = et.data.get_data('cold-springs-fire')
os.chdir(os.path.join(et.io.HOME, 'earth-analytics'))

# Stack the Landsat pre fire data
landsat_paths_pre = glob(
    "data/cold-springs-fire/landsat_collect/LC080340322016070701T1-SC20180214145604/crop/*band*.tif")
path_landsat_pre_st = 'data/cold-springs-fire/outputs/landsat_pre_st.tif'

landsat_paths_pre.sort()
es.stack(landsat_paths_pre, path_landsat_pre_st)

# Read landsat pre fire data
with rio.open(path_landsat_pre_st) as landsat_pre_src:
    landsat_pre = landsat_pre_src.read(masked=True)
    landsat_extent = plotting_extent(landsat_pre_src)

# Open the pixel_qa layer for your landsat scene
with rio.open("data/cold-springs-fire/landsat_collect/LC080340322016070701T1-SC20180214145604/crop/LC08_L1TP_034032_20160707_20170221_01_T1_pixel_qa_crop.tif") as landsat_pre_cl:
    landsat_qa = landsat_pre_cl.read(1)
    landsat_ext = plotting_extent(landsat_pre_cl)

# Generate array of all possible cloud / shadow values
cloud_shadow = [328, 392, 840, 904, 1350]
cloud = [352, 368, 416, 432, 480, 864, 880, 928, 944, 992]
high_confidence_cloud = [480, 992]

all_masked_values = cloud_shadow + cloud + high_confidence_cloud

# Call the earthpy mask function using pixel QA layer
landsat_pre_cl_free = em.mask_pixels(landsat_pre, landsat_qa, vals=all_masked_values)
````

This code is taken from the [Earth Analytics Python week 8 lesson](https://www.earthdatascience.org/courses/earth-analytics-python/multispectral-remote-sensing-modis/cloud-masks-with-spectral-data-python/).

**Expected behavior**
When the parameters are passed to the `_create_mask` helper function, a mask should be created in the areas in the pixel qa layer that contain the values passed in.

**What Operating System Are you Running?**
 - Windows

**Comment by mirob9363** [comment]
@lwasser can you take a peak at this? There must be some issue with how the `_create_mask` helper function is checking to see if the values are in the pixel qa layer.

**Comment by lwasser** [comment]
This is really unusual behavior. Essentially what is happening is _create_mask is MODIFYING the input parameter. This doesn't make sense to me but as a fix i ensured that a copy of the input array was created in the function to avoid any intentional modifications to the original object.

**Comment by lwasser** [comment]
ok - i have looked into this and the behavior that we are seeing is expected!! i have implemented a fix and merged it via pr #277 referenced above. closing this for the time being!!
